File: source/core/src/main/java/com/padr/tuxdb/server/controller/DatabaseController.java
package com.padr.tuxdb.server.controller;
import com.padr.tuxdb.engine.process.pub.Database;

import spark.Request;

public class DatabaseController {

    public static Object handler(String function, Request request) {
        try {
            switch (function) {
                case "getDatabaseNames":
                    return Database.getDatabaseNames();
                case "createDatabase":
                    return Database.createDatabase(request.queryParams("databaseName"));
                case "getCollectionNames":
                    return new Database(request.queryParams("databaseName")).getCollectionNames();
                case "setDatabaseName":
                    return new Database(request.queryParams("databaseName")).setDatabaseName(request.queryParams("newDatabaseName"));
                case "isExist":
                    return new Database(request.queryParams("databaseName")).isExist();
                case "drop": 
                    return new Database(request.queryParams("databaseName")).drop();
                default:
                    return null;
            }
        } catch (Exception e) {
            return null;
        }
    }
    
}

File: tools/shell/src/controller/databaseController.py
import json

from src.connection import connection
from src.environmentVariable import EnvironmentVariable
from src.constant import Constant


class DatabaseController:

    __SERVICE_NAME = "database"

    @staticmethod
    def getDatabaseNames():
        print(connection.send(
            DatabaseController.__SERVICE_NAME, "getDatabaseNames", {}))

    @staticmethod
    def createDatabase():
        print(connection.send(DatabaseController.__SERVICE_NAME,
              "createDatabase", {"databaseName": EnvironmentVariable.databaseName}))

    @staticmethod
    def getCollectionNames():
        print(connection.send(DatabaseController.__SERVICE_NAME,
              "getCollectionNames", {"databaseName": EnvironmentVariable.databaseName}))

    @staticmethod
    def setDatabaseName():
        newDatabaseName = input(
            f"{Constant.COMMAND_QUESTION_PREFIX}New Database Name: ")

        print(connection.send(DatabaseController.__SERVICE_NAME, "setDatabaseName", {
              "databaseName": EnvironmentVariable.databaseName, "newDatabaseName": newDatabaseName}))

    @staticmethod
    def drop():
        print(connection.send(DatabaseController.__SERVICE_NAME,
              "drop", {"databaseName": EnvironmentVariable.databaseName}))
